### PR TITLE
ansible: Replace nebula with Wireguard tunnels

### PR DESCRIPTION
**Pull request by the-maldridge**
The tunnels are still densely meshed and should allow point to point communications between all hosts.

@steinex please review.

### CODE REVIEW

File: ansible/roles/void-mesh/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    pkg: wireguard-tools
+    state: present
+
+- name: Install Config
+  template:
+    src: void0.conf.j2
+    dest: /etc/wireguard/void0.conf
+    owner: root
+    group: root
+    mode: 0644
```
Comment (steinex) [inline ansible/roles/void-mesh/tasks/main.yml:13]: doesn't need to be world readable

**Review by steinex — COMMENTED** [review]
otherwise LGTM :+1:

**Comment by zdykstra** [comment]
I've not used Wireguard before, but the steps taken here seem sane. `secrets/vpn` already ignores the files in there, so there won't be an accidental leak of wireguard keys.

File: docs/src/network/index.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Void's global network is based on a Wireguard mesh between all
```
Comment (ericonr) [inline docs/src/network/index.md:1]: Perhaps add a `# Network` to the top?

File: docs/src/services/vpn.md [hunk 3]
```diff
@@ -1,11 +0,0 @@
...
-
-Void maintains an overlay mesh network which contains all of Void's
-hosts.  This network uses the `nebula` VPN implementation to provide
-point-to-point overlay tunnels across the internet.
-
-Access to this network is tightly controlled, and is provided on a
-need-to-use basis.  Typically membership is restricted to
-`netauth/dante` and even then the VPN links should only be activated
-when necessary.  A strong preference should be shown to accessing
-systems via HTTP transports over the internet.
```
Comment (ericonr) [inline docs/src/services/vpn.md:11]: Is this part still true? Perhaps it should be added to the network page as well.

File: docs/src/network/index.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+servers.  The network does not support ECMP.  The mesh is statically
+computed by Ansible and is installed into the fleet when machines are
+added or removed.
+
+The primary use case of the mesh is to allow us to run services that
+expect to be inside of a single broadcast domain without expending
+significant effort to enable machine to machine communication.
+
```
Comment (ericonr) [inline docs/src/network/index.md:9]: Trailing new line.

**Pull request merged**